app/controllers/api/proposals_controller.rb

```diff
@@ -34,8 +35,19 @@ def show
     render json: @proposal
   end
 
+  def update
```

Comment:
This could be checked in the `strong_params` part, and you could completely skip `can? :update, Proposal` as it would be using an empty strong params list.